java/src/main/java/com/logicalclocks/hsfs/engine/FeatureGroupEngine.java

```diff
@@ -174,8 +172,20 @@ public StreamingQuery insertStream(FeatureGroup featureGroup, Dataset<Row> featu
           + "`, with version `" + featureGroup.getVersion() + "` will not perform validation.");
     }
 
-    return SparkEngine.getInstance().writeStreamDataframe(featureGroup, utils.sanitizeFeatureNames(featureData),
-        queryName, outputMode, awaitTermination, timeout, getKafkaConfig(featureGroup, writeOptions));
+    StreamingQuery streamingQuery = SparkEngine.getInstance().writeStreamDataframe(featureGroup,
+        utils.sanitizeFeatureNames(featureData), queryName, outputMode, awaitTermination, timeout,
+        utils.getKafkaConfig(featureGroup, writeOptions));
+
+    // start streaming to hudi table from online feature group topic
+    if (featureGroup.getTimeTravelFormat() == TimeTravelFormat.HUDI) {
+      writeOptions.put("functionType", "streamingQuery");
```

Comment:
Yes but how does hopsworks know if this is `insert` or `insert_stream`? Allright here is the problem: 1st time feature group is created with `save` we will save data as before. next time when `insert` or `insert_stream` is called It will change the source to Kafka topic, reset checkpoint from there and it will be always DeltaStreamer with kafka source.  We can't just change sources back and force.          

Problem is after the `insert` we immediately call `compute_statistics` read commit metadata data information etc.  And considering that DeltaStreamer job may not start immediately then I don't know how what the best way to handle it